components/file_upload/file_upload.test.tsx

```diff
@@ -367,42 +374,44 @@ describe('components/FileUpload', () => {
         const pluginHook = () => {
             return {files: null};
         };
-        const props = {...baseProps, pluginFilesWillUploadHooks: [{hook: pluginHook}]};
-        const files = [{name: 'file1.pdf'}, {name: 'file2.jpg'}];
+        const props = {...baseProps, pluginFilesWillUploadHooks: [{hook: pluginHook} as unknown as PluginComponent]};
+        const files = [{name: 'file1.pdf'} as File, {name: 'file2.jpg'} as File];
 
         const wrapper = shallowWithIntl(
             <FileUpload {...props}/>,
         );
 
-        wrapper.instance().checkPluginHooksAndUploadFiles(files);
+        const instance = wrapper.instance() as FileUploadClass;
+        instance.checkPluginHooksAndUploadFiles(files);
 
         expect(uploadFile).toHaveBeenCalledTimes(0);
 
-        expect(baseProps.onUploadStart).toHaveBeenCalledTimes(0);
+        expect(baseProps.onUploadStart).toHaveBeenCalledTimes(1);
```

Comment:
I updated this but I still feel safe because even though `onUploadStart` is called `onUploadError` is called afterword. Therefore I think the test is passing without changing its initial logic (i.e.  FilesWillUploadHook - should reject all files)

Reply:
I think this might actually be a change we care about. I think it's happening because of [this line](https://github.com/mattermost/mattermost-webapp/pull/10858/files#diff-df80b22dab3168acc877a03b6dbacaaa5509a191b4815cd5ecb3e31e4bdea53dR246) (the line that changed to `sortedFiles = result?.files || [];` in case that link doesn't work.

When the hook returns an error, we're now setting `sortedFiles` to an array, so the if statement that controls the call to `onUploadStart` doesn't prevent this call any more.

Thanks for calling out this change since we might've missed it otherwise

Reply:
Exactly, that line you men is causing the change in the test. 

Do you expect me to change the if statement to check if `sortedFiles` is empty array, so we don't call `onUploadStart `? Or updating the test is here enough?

Reply:
Yeah, I think that'd be the best way to do this rather than changing the test. I'd be worried that having `onUploadStart` called even when an error occurs might cause some other logic to change

Reply:
Okay, it is updated as suggested